base/poco/Foundation/src/Thread_POSIX.cpp

```diff
@@ -197,9 +200,14 @@ void ThreadImpl::startImpl(SharedPtr<Runnable> pTarget)
 	_pData->pRunnableTarget = pTarget;
 	if (pthread_create(&_pData->thread, &attributes, runnableEntry, this))
 	{
-		_pData->pRunnableTarget = 0;
-		pthread_attr_destroy(&attributes);
-		throw SystemException("cannot start thread");
+		FastMutex::ScopedLock l(_pData->mutex);
+		_pData->pRunnableTarget = pTarget;
+		if (pthread_create(&_pData->thread, &attributes, runnableEntry, this))
+		{
+			_pData->pRunnableTarget = 0;
+			pthread_attr_destroy(&attributes);
+			throw SystemException("cannot start thread");
+		}
```

Comment:
This can't be right. There are two `pthread_create` there, the first one not protected by the mutex. I think something got mixed up when bringing the changes. The original code is not like that: https://github.com/pocoproject/poco/blob/e2c2899d4e47610c831c8dd2215c2b702f9fc98b/Foundation/src/Thread_POSIX.cpp#L324-L334

Reply:
You are right. I made a mistake when copying the fix